packages/astro/components/Code.astro

```diff
@@ -92,7 +93,13 @@ if (typeof lang === 'object') {
 }
 
 const highlighter = await getCachedHighlighter({
-	langs: [lang],
+	langs: [
+		typeof lang === 'string'
+			? Object.keys(bundledLanguages).includes(lang)
+				? lang
+				: 'plaintext'
+			: lang,
+	],
```

Comment:
This seems okay, but I think the ideal solution would be to not pass it through the highlighter at all if we don't detect a language.

Reply:
I'm not familiar with shikiji, so it langs optional or de we need to pass an empty array?

Reply:
Looking at https://github.com/withastro/astro/blob/main/packages/markdown/remark/src/shiki.ts#L28, I suppose both work since it will an empty array by default but a confirmation wouldn't hurt!

Reply:
I think a fix like now should be fine. We need shikiji to output the specific code/pre HTML elements, so not passing through the highlighter would be a little tricky.

Reply:
I'm cool with this, then! Makes sense.